### PR TITLE
feat(coco): adds support for bounding boxes (also rotatable)

### PR DESCRIPTION
**Pull request by Encord-davids**

### CODE REVIEW

File: src/encord_active/app/data_quality/sub_pages/explorer.py [hunk 1]
```diff
@@ -138,9 +138,9 @@ def fill_data_quality_window(
     if (embedding_information.type == EmbeddingType.CLASSIFICATION) and len(embedding_information.collections) == 0:
         st.write("Image-level embedding file is not available for this project.")
         return
-    if (embedding_information.type == EmbeddingType.OBJECT) and len(embedding_information.collections) == 0:
-        st.write("Object-level embedding file is not available for this project.")
-        return
+    # if (embedding_information.type == EmbeddingType.OBJECT) and len(embedding_information.collections) == 0:
+    #     st.write("Object-level embedding file is not available for this project.")
+    #     return
```
Comment (yogesh-encord) [inline src/encord_active/app/data_quality/sub_pages/explorer.py:143]: Remove if not useful?

File: src/encord_active/lib/metrics/metric.py [hunk 2]
```diff
@@ -8,44 +8,29 @@
...
 
 
-class DataType(Enum):
+class DataType(str, Enum):
     IMAGE = "image"
     SEQUENCE = "sequence"
 
 
-class EmbeddingType(Enum):
+class EmbeddingType(str, Enum):
```
[the remark this answers is not in the document]
Comment (Encord-davids) [reply src/encord_active/lib/metrics/metric.py:27]: They would 💯

File: docs/docs/metrics/write-your-own.md [hunk 3]
```diff
@@ -5,12 +5,12 @@ sidebar_position: 5
 # Write Your Own Metric
 
 :::tip
-If you are more comfortable with the notebooks, we have a 
-[Google Colab notebook](https://colab.research.google.com/drive/1tAqGGSY0sZfwec2Vp4ThvgLKIefy3-4b?usp=sharing) 
-for writing your own metric.  
+If you are more comfortable with the notebooks, we have a
```
[the remark this answers is not in the document]
Comment (Encord-davids) [reply docs/docs/metrics/write-your-own.md:8]: makes sense

File: src/encord_active/lib/common/image_utils.py [hunk 4]
```diff
@@ -189,6 +192,16 @@ def __get_geometry(obj: dict, img_h: int, img_w: int) -> Optional[Tuple[str, np.
                 [b["x"] * img_w, (b["y"] + b["h"]) * img_h],
             ]
         )
+    elif obj["shape"] == ObjectShape.ROTATABLE_BOUNDING_BOX:
+        b = obj["rotatableBoundingBox"]
+        top_left = [b["x"] * img_w, b["y"] * img_h]
+        top_right = [(b["x"] + b["w"]) * img_w, b["y"] * img_h]
+        bottom_right = [(b["x"] + b["w"]) * img_w, (b["y"] + b["h"]) * img_h]
+        bottom_left = [b["x"] * img_w, (b["y"] + b["h"]) * img_h]
+        rotated_polygon = rotate(Polygon([top_left, top_right, bottom_right, bottom_left]), b["theta"])
```
[the remark this answers is not in the document]
Comment (Encord-davids) [reply src/encord_active/lib/common/image_utils.py:201]: good point 👍

File: src/encord_active/lib/coco/utils.py [hunk 5]
```diff
@@ -80,15 +81,23 @@ def make_object_dict(
 
         object_dict[shape] = {"0": {"x": round(object_data[0], 4), "y": round(object_data[1], 4)}}
 
-    elif shape == "bounding_box":
+    elif shape in BoxShapes:
         if not isinstance(object_data, dict):
             raise ValueError(f"The `object_data` for {shape} should be a dictionary.")
-        if len(BBOX_KEYS.union(set(object_data.keys()))) != 4:
+        if len(BBOX_KEYS.intersection(object_data.keys())) != 4:
+            __import__("ipdb").set_trace()
```
[the remark this answers is not in the document]
Comment (Encord-davids) [reply src/encord_active/lib/coco/utils.py:88]: good catch 🙈

File: src/encord_active/app/data_quality/sub_pages/explorer.py (hunk 1 above)
Comment (Encord-davids) [reply src/encord_active/app/data_quality/sub_pages/explorer.py:187]: it is actually an important part, I commented it out during some of my tests.

**Review by yogesh-encord — APPROVED** [review]

**Pull request merged**